### PR TITLE
Troubleshooting: Refactor related problems

### PR DESCRIPTION
**Pull request by andyg0808**
This should be a pure refactor. The goal is to reduce the amount of code which is affected by implementing the frontend for https://github.com/iFixit/ifixit/issues/50604.

## CR Notes
Turn on "ignore whitespace" on the diff; this is just extracting a couple functions and adding a new key to a type.

## QA Notes
Please check that the `Related Problems` section of the page still looks reasonable.

Connects https://github.com/iFixit/ifixit/issues/50604

### CODE REVIEW

**Comment by deltuh-vee** [comment]
QA 🎬 
Related Problems section still looks fine.

File: frontend/templates/troubleshooting/hooks/useTroubleshootingProps.tsx [hunk 1]
```diff
@@ -89,6 +89,7 @@ export type TroubleshootingApiData = {
    metaDescription: string;
    metaKeywords: string;
    linkedProblems: Problem[];
+   relatedProblems: Problem[];
```
Comment (jyee27) [inline frontend/templates/troubleshooting/hooks/useTroubleshootingProps.tsx:92]: When/where did this get added to the API?

**Review by jyee27 — APPROVED** [review]
CR ☕

File: frontend/templates/troubleshooting/hooks/useTroubleshootingProps.tsx (hunk 1 above)
Comment (andyg0808) [reply frontend/templates/troubleshooting/hooks/useTroubleshootingProps.tsx:92]: https://github.com/iFixit/ifixit/pull/50620

It's not gone out yet, apparently. Nothing's using the data, though.

**Pull request merged**